docker/api/build.py

```diff
@@ -11,7 +12,47 @@
 
 log = logging.getLogger(__name__)
 
-
+class FileUploadBuffer(object):
+    '''
+    Wrap a file object, buffer, and print progress.
+    '''
+    BUFFER_SIZE = 1024*32
+    def __init__(self, file):
+        self.read_so_far = 0
+        self.num_reads = 0
+        self.starttime = datetime.datetime.utcnow()
+        self.file = file
```

Comment:
- I can remove total reads, I initially just used it to properly size the buffer to something reasonable, but it's probably not necessary anymore.
- As it stands, this function does not rely on fseek or ftell, which would be required to implement the file size. Not sure we want to add that dependency.